### ISSUE TITLE
Rule proposal: Improvement for filename-case rule

### ISSUE THREAD

**Issue by DavidFontalba** [sindresorhus/eslint-plugin-unicorn#1745]
### Description

It would be great to be able to specify a path for each naming convention for the filename-case rule. We'd love to have this behaviour because in our project we have two main file extensions (`.ts` and `.spec.ts`) and each one use a different case: camelCase for first ones and snakeCase for second ones.

As the actual configuration for the rule is in object like:
```js
"unicorn/filename-case": [
	"error",
	{
		"case": "kebabCase",
		"ignore": [
			"\\.SOMETHING\\.js$",
		]
	}
]
```

We propose to replace the object by an array that contains several objects. So, the rule could maintain the actual behaviour if needed and also could accept different objects to resolve our problem.
```js
"unicorn/filename-case": [
	"error",
	[
		{
			"case": "camelCase",
			"ignore": [
				".spec.ts",
			]
		},
		{
			"case": "snakeCase",
			"ignore": [
				".ts",
			]
		}
	]
]
```

Optional: After these proposed changes it will be more logical to have an 'include' option. So, it could work like the following example:
```js
"unicorn/filename-case": [
	"error",
	[
		{
			"case": "camelCase",
			"include": [
				".ts",
			]
		},
		{
			"case": "snakeCase",
			"include": [
				".spec.ts",
			]
		}
	]
]
```

### Fail

Using the previous example configuration, the following names will fail:
```js
someSpecFile.spec.ts // Should be snakeCase
non_spec_file.ts // Should be camelCase
OtherFiles.ts // Should be camelCase
OtherFiles.spec.ts // Should be snakeCase
```

### Pass

Using the previous example configuration, the following names will pass:
```js
some_spec_file.spec.ts
nonSpecFile.ts
otherFiles.ts
other_files.spec.ts
```

**Comment by sindresorhus** [comment]
Any reason you cannot use the ESLint overrides config?

**Comment by DavidFontalba** [comment]
We didn't know about this feature of ESLint, it's exactly what we were looking for. Thanks for the help 😁
